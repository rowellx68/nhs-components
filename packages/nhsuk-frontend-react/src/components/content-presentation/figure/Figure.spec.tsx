import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  ImageWithCaption as ImageWithCaptionStory,
  ImageWithoutCaption as ImageWithoutCaptionStory,
} from './Figure.stories';

const ImageWithCaption = composeStory(ImageWithCaptionStory, meta);
const ImageWithoutCaption = composeStory(ImageWithoutCaptionStory, meta);

it('should render the Figure component', () => {
  const { container } = render(<ImageWithCaption />);

  expect(container).toMatchSnapshot();
});

it('should render the Figure component without caption', () => {
  const { container } = render(<ImageWithoutCaption />);

  expect(container).toMatchSnapshot();
});
