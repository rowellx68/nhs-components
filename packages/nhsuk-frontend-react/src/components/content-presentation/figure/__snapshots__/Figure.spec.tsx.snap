// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the Figure component 1`] = `
<div>
  <figure
    class="nhsuk-image"
  >
    <img
      alt="Lots of sore red patches with small blisters spread across white skin on a woman's chest."
      class="nhsuk-image__img"
      src="https://assets.nhs.uk/prod/images/S_0318_Bullous_pemphigoid_lesions_.2e16d0ba.fill-320x213.jpg"
    />
    <figcaption
      class="nhsuk-image__caption"
    >
      It can affect large areas of the body or limbs.
    </figcaption>
  </figure>
</div>
`;

exports[`should render the Figure component without caption 1`] = `
<div>
  <figure
    class="nhsuk-image"
  >
    <img
      alt="Lots of sore red patches with small blisters spread across
        white skin on a woman's chest."
      class="nhsuk-image__img"
      src="https://assets.nhs.uk/prod/images/S_0318_Bullous_pemphigoid_lesions_.2e16d0ba.fill-320x213.jpg"
    />
  </figure>
</div>
`;
